### ISSUE TITLE
Introduction of tasks for dismissing alarms

### ISSUE THREAD

**Issue by Rafee-M** [FossifyOrg/Clock#47]
### Checklist

- [X] I made sure that there are **no existing issues** - [open](https://github.com/FossifyOrg/Clock/issues) or [closed](https://github.com/FossifyOrg/Clock/issues?q=is%3Aissue+is%3Aclosed) - to which I could contribute my information.
- [X] I made sure that there are **no existing discussions** - [open](https://github.com/FossifyOrg/Clock/discussions) or [closed](https://github.com/FossifyOrg/Clock/discussions?discussions_q=is%3Aclosed) - to which I could contribute my information.
- [X] I have read the FAQs inside the app (Menu -> About -> FAQs) and my problem isn't listed.
- [X] **I have taken the time to fill in all the required details. I understand that the bug report will be dismissed otherwise.**
- [X] This issue contains only one feature request.
- [X] I have read and understood the [contribution guidelines](https://github.com/FossifyOrg/Clock/blob/master/CONTRIBUTING.md).
- [ ] I optionally [donated](https://fossify.org/donate) to support the Fossify mission.

### Feature description

I think it would be helpful if math problems were added as a barrier to dimissing alarms. 

### Why do you want this feature?

This feature would greatly increase the effectiveness of waking someone up I believe

**Comment by chesio** [comment]
Or maybe just get more sleep ;-) Don't get me wrong, it's an interesting idea, but I'm not sure how many people would really use it. Personally, I would prefer that Fossify apps stay simple and avoid feature creep.

**Comment by naveensingh** [comment]
I agree with @chesio. I have tried those alarms for a couple of months and it wasn't a great couple of months. Sleep deprivation is not good.

**Comment by LeonelRFF** [comment]
For me it is also a good function, I don't find it hard to get up but others do and this is useful for them.

**Comment by TBunnie** [comment]
I would love to see similar features implemented as well. I'd prefer scanning a bar code over a math problem, (it insures I can't dismiss the alarm without taking my medication) but anything to prevent me from shutting the alarm off and going back to sleep would be helpful.

**Comment by matheuswillder** [comment]
Almost a year later, but since this issue still open, I decided to bring here the screenshot I posted from an app that had this feature in the duplicate issue that I accidentally made https://github.com/FossifyOrg/Clock/issues/172, just as a reference in case it's ever implemented.

Unfortunately, the app is no longer available and doesn't work on current versions of Android.

And, even more unfortunately, I still have trouble waking up (I've already accepted that, at this point in life, I'll probably have it for the rest of my life) lol

<img alt="ignoreImageMinify" src="https://github.com/user-attachments/assets/b49df0cb-ed20-48c1-84d5-58fd7e6d33a6" width=225 />
